examples/step-83/doc/intro.dox

```diff
@@ -0,0 +1,608 @@
+do this:
+@code
+template <int rank, int dim, typename Number>
+class Tensor
+{
+public:
+  //
+  // Read or write the data of this object to or from a stream for the purpose
+  // of serialization using the BOOST serialization
+  // library.
+  //
+  template <class Archive>
+  void
+  serialize(Archive &ar, const unsigned int version);
+
+private:
+  //
+  // Array of tensors holding the subelements.
+  //
+  Tensor<rank-1, dim, Number> values[dim];
+};
+
+
+template <int rank, int dim, typename Number>
+template <class Archive>
+inline void
+Tensor<rank, dim, Number>::serialize(Archive &ar, const unsigned int /*version*/)
+{
+  ar & values;
+}
+@endcode
+
+In other words, the Tensor class stores an array of `dim` objects of a
+tensor of one rank lower, and in order to serialize this array, the
+function simply uses the overloaded `operator&` which recognizes that
+`values` is a C-style array of fixed length, and what the data type of
+the elements of this array are. It will then in turn call the
+`serialize()` member function of `Tensor<rank-1,dim>`. (The recursion
+has a stopping point which is not of importance for this discussion. The
+actual implementation of class Tensor looks different, but you get the
+general idea here.)
+
+Depending on whether the `Archive` type used for the first argument of
+this function is an input or output archive, `operator&` reads from or
+writes into the `values` member variable. If the class had more member
+variables, one would list them one after the other. The same is true
+if a class has base classes. For example, here is the corresponding
+function for the Quadrature class that stores quadrature points and
+weights, and is derived from the Subscriptor base class:
+@code
+template <int dim>
+template <class Archive>
+inline void
+Quadrature<dim>::serialize(Archive &ar, const unsigned int)
+{
+  // Forward to the (de)serialization function in the base class:
+  ar & static_cast<Subscriptor &>(*this);
+
+  // Then (de)serialize the member variables:
+  ar & quadrature_points & weights;
+}
+@endcode
+The beauty of having to write only one function for both serialization
+and deserialization is that obviously one has to read data from the
+archive in the same order as one writes into it, and that is easy
+to get wrong if one had to write two separate functions -- but that
+is automatically correct if one has the same function for both
+purposes!
+
+The BOOST serialization library also has mechanisms if one wants to
+write separate save and load functions. This is useful if a class
+stores a lot of data in one array and then has a cache of commonly
+accessed data on the side for fast access; one would then only store
+and load the large array, and in the load function re-build the cache
+from the large array. (See also the discussion below on what one
+actually wants to save.)
+
+Based on these (de)serialization functions, one would then first write a
```

Comment:
The transition here is very abrupt. Until now, you talked generally about serialization/deserialization and suddenly you talk about the concrete example without introducing the concrete example.